src/components/posts/postForm.js

```diff
@@ -0,0 +1,81 @@
+import React, { useContext, useEffect, useRef } from "react"
+import Form from 'react-bootstrap/Form';
+import FormGroup from 'react-bootstrap/FormGroup';
+import Button from 'react-bootstrap/Button';
+// import NewPostButton from "./NewPostButton";
+import { CategoryContext } from "../categories/CategoryProvider";
+import { PostContext } from "./PostProvider";
+export const PostForm = (props) => {
+    const {createPost} = useContext(PostContext)
+    const {categories, getCategories} = useContext(CategoryContext)
+    
+
+    const titleRef = useRef("")
+    const categoryRef = useRef("")
+    const imageRef = useRef("")
+    const contentRef = useRef("")
+    const publicationRef = useRef("")
+    const constructNewPost = () => {
+
+        
+        if (categoryRef.current.value === '0') {
+            window.alert("Please select a category")
+        } else {
+        createPost({
+            user_id: localStorage.getItem('rare_user_id'),
+            title: titleRef.current.value,
+            category_id: categoryRef.current.value,
+            image: imageRef.current.value,
+            content: contentRef.current.value,
+            publish_status: true,
+            approve_status: true,
+            publication_time: publicationRef.current.valueAsNumber,
+            creation_time: Date.now()
+
+        })
+        .then((newPost) => props.history.push(`/posts/${newPost.id}`))
+    }
+    }
+
+    useEffect(()=>{
+        getCategories()
+    },[])
+
+    return (
+        <Form>
+            <FormGroup controlId ="title">
+                <Form.Label>Title</Form.Label>
+                <Form.Control type="text" placeholder="Enter Title" ref={titleRef} />
+            </FormGroup>
+            <Form.Group controlId="categorySelect">
+            <Form.Label>Categories</Form.Label>
+                <Form.Control as="select" ref={categoryRef}>
+                <option value ="0">Select a category</option>
+                {categories.map(c => (
+                    <option key={c.id} value={c.id}>
+                        {c.name}
+                    </option>
+                ))}
+                </Form.Control>
+            </Form.Group>
+            <FormGroup>
+                <Form.Label>Publish Date:</Form.Label>
+                <Form.Control type="date" ref={publicationRef}/> 
+            </FormGroup>
+            <FormGroup controlId ="Text">
+                <Form.Label>Image</Form.Label>
+                <Form.Control type="Text" placeholder="Image Url" ref={imageRef} />
+            </FormGroup>
+            <FormGroup controlId ="Text">
+                <Form.Label>Content</Form.Label>
+                <Form.Control as="Textarea" rows={3} placeholder="Enter post..." ref={contentRef} />
```

Comment:
It seems like react-bootstrap doesn't like you giving "Textarea" a capital T, since `<textarea>` is a builtin HTML component and thus should be lower-case, and not a custom React component that would be uppercase. Changing this to lower case should get rid of this error in the console